File: cpp/find_minimum_in_rotated_sorted_array_ii.cc
/*
 * Filename: find_minimum_in_rotated_sorted_array.cc
 * Author: Jun Xu
 * Email: jxu.vimacs@gmail.com
 * Created Time: Fri 07 Aug 2015 06:36:59 PM EDT
 */

#include <iostream>
using namespace std;

class Solution {
	public:
		int findMin(vector<int>& nums)
		{
			if (nums.empty())
			{
				return 0;
			}

			for (vector<int>::size_type i = 0; i < nums.size(); i++)
			{
				if (i && nums[i] < nums[i-1])
				{
					return nums[i];
				}
			}

			return nums[0];
		}
};

File: java/FindMinimuminRotatedSortedArrayII.java
/*
 * Filename: FindMinimuminRotatedSortedArrayII.java
 * Author: Jun Xu
 * Email: jxu.vimacs@gmail.com
 * Created Time: Wed 22 Jul 2015 03:16:36 PM EDT
 */

public class Solution {
	public int findMin(int[] nums) {
		if (nums == null || nums.length == 0) {
			return 0;
		}

		if (nums.length == 1) {
			return nums[0];
		}

		for (int i = 0; i < nums.length; i++) {
			if (i < nums.length - 1 && nums[i] > nums[i+1]) {
				return nums[i+1];
			}
		}

		return nums[0];
	}
}
